crates/miden-testing/src/kernel_tests/tx/test_account.rs

```diff
@@ -1123,3 +1114,66 @@ fn transaction_executor_account_code_using_custom_library() -> miette::Result<()
     );
     Ok(())
 }
+
+/// Tests that incrementing the account nonce twice fails.
+#[test]
+fn incrementing_nonce_twice_fails() -> anyhow::Result<()> {
+    let source_code = "
+        use.miden::account
+
+        export.auth__incr_nonce_twice
+            exec.account::incr_nonce
+            exec.account::incr_nonce
+        end
+    ";
+
+    let faulty_auth_component =
+        AccountComponent::compile(source_code, TransactionKernel::assembler(), vec![])?
+            .with_supports_all_types();
+    let (account, seed) = AccountBuilder::new([5; 32])
+        .with_auth_component(faulty_auth_component)
+        .with_component(
+            AccountMockComponent::new_with_empty_slots(TransactionKernel::assembler()).unwrap(),
+        )
+        .build()
+        .context("failed to build account")?;
+
+    let err = TransactionContextBuilder::new(account)
+        .account_seed(Some(seed))
+        .build()?
+        .execute()
+        .unwrap_err();
+
+    let TransactionExecutorError::TransactionProgramExecutionFailed(err) = err else {
+        anyhow::bail!("expected TransactionExecutorError::TransactionProgramExecutionFailed");
+    };
+
+    assert_execution_error!(Err::<(), _>(err), ERR_ACCOUNT_NONCE_CAN_ONLY_BE_INCREMENTED_ONCE);
+
+    Ok(())
+}
+
+/// Tests that incrementing the account nonce fails if it would overflow the field.
+#[test]
+fn incrementing_nonce_overflow_fails() -> anyhow::Result<()> {
+    let mut account = AccountBuilder::new([42; 32])
+        .with_auth_component(Auth::IncrNonce)
+        .with_component(
+            AccountMockComponent::new_with_empty_slots(TransactionKernel::assembler()).unwrap(),
+        )
+        .build_existing()
+        .context("failed to build account")?;
+    // Increment the nonce to the maximum felt value. The nonce is already 1, so we increment by
+    // modulus - 2.
+    account.increment_nonce(Felt::new(Felt::MODULUS - 2))?;
```

Comment:
This prompts to think: do we still need the `nonce_delta` parameter on `Account::implement_nonce`? It would make the overflow very difficult (or long) to test for sure :) but I think we should be consistent with other places where we remove this argument.

Reply:
Regarding this comment and this one:
> The comment could probably be adjusted (and the struct's doc comment too)

Not completely sure what you mean. The `nonce_delta` in `AccountDelta` still can technically be any felt, because it is not only used to represent the delta of a single transaction but also of multiple transactions (using `AccountDelta::merge`). So its value can be 3 if three transactions have been merged, and so `increment_nonce` must take a parameter so the nonce can be incremented by 3. Does that make sense, or did I misunderstand?